tests/TestCase/ORM/RulesCheckerIntegrationTest.php

```diff
@@ -490,6 +490,28 @@ public function testIsUniqueMultipleFieldsOneIsNull()
     }
 
     /**
+     * Tests the IsNotReferencedBy rule before delete
+     *
+     * @group delete
+     * @return void
+     */
+    public function testIsNotReferencedByRule()
+    {
+        $articlesTable = TableRegistry::get('Articles');
+        $articlesTable->belongsTo('Authors');
+        $authorsTable = TableRegistry::get('Authors');
+
+        $author = $authorsTable->get(1);
+        $authorWithoutArticles = $authorsTable->get(2);
+
+        $rules = $authorsTable->rulesChecker();
+        $rules->addDelete($rules->IsNotReferencedBy('Articles'));
```

Comment:
@markstory just a quick question here, it seems to me that with each Rule added, we need to add a method to the RulesChecker. (Seems like this might be boilerplate code) Would it be preferable to modify the internals of RulesChecker to automatically load rules from the `Cake\ORM\Rule` namespace? It could be implemented easily as a single magic method.